### PR TITLE
Fixes missing icon in PM UI due to backlash icon entry

### LINKED ISSUE #9225: [Test Failure] Embedded icon can’t be shown in PM UI with local package feed - depending on / vs \
## Details about Problem
NuGet version: NuGet Client Dev\5.6.0.6472
VS Version: Master\29825.171
OS: Windows-10-Enterprise-RS5
## Detailed repro steps
1.	Create a Class Library (.NET Framework 4.8) in VS and build the solution.
2.	In the Developer Command Prompt for VS, create a nuspec file by executing command “nuget spec”
3.  Make an image file (size limited to 1 MB with JPEG/PNG format and 128x128 resolution) and locate in the package.
4.  Add following to nuspec file.
```
<package>
  <metadata>
    ...
    <icon>images\icon.png</icon>
    ...
  </metadata>
  <files>
    ...
    <file src="..\icon.png" target="images\" />
    ...
  </files>
</package>
```
5.  Pack the package to local path by executing command “nuget pack”.
6.  Create a C# Console App (.NET Framework 4.8) in VS and open the PM UI of the project.
7.  Add the package source as a local feed in Settings->NuGet Package Manager->Package Sources.
8.  Select the new added local feed in the Package source dropdown box in the PM UI and install the package.

## Expected
The embedded icon show correctly in all the tabs of PM UI.
## Actual
The embedded icon can’t be shown in all the tabs of PM UI. as below screenshot.
image
## Note
1.	It doesn’t repro for the package creating by “dotnet pack” with local package feed.
2.	It doesn’t repro for the package “Text2Xml.Lib” (version 1.1.1 and 1.1.2) in remote feed.
3.	It also repro with icon.jpg.

### PR DESCRIPTION
**Pull request by dominoFire**
## Bug

Fixes: https://github.com/NuGet/Home/issues/9225
Regression: Yes?
* Last working version: 16.5 p3
* How are we preventing it in future: Testing with corner cases.

## Fix

Details: When converting an embedded Icon URI, it normalizes the icon entry from internal nuspec.

## Testing/Validation

Tests Added: Yes. Updated tests.
Validation: Manual run.

Note: This fix only works with packages produced with NuGet

### CODE REVIEW

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Converters/IconUrlToImageCacheConverter.cs [hunk 1]
```diff
@@ -80,6 +81,7 @@ public object Convert(object[] values, Type targetType, object parameter, Cultur
             if (IsEmbeddedIconUri(iconUrl))
             {
                 var iconEntry = Uri.UnescapeDataString(iconUrl.Fragment).Substring(1); // skip the '#' in a URI fragment
+                var iconEntryNormalized = PathUtility.StripLeadingDirectorySeparators(iconEntry) ;
```
Comment (cristinamanum) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/Converters/IconUrlToImageCacheConverter.cs]: is it an extra space before ";"?

**Review by cristinamanum — APPROVED** [review]

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Converters/IconUrlToImageCacheConverter.cs (hunk 1 above)
Comment (nkolev92) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/Converters/IconUrlToImageCacheConverter.cs]: is iconEntry used anywhere now?

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Converters/IconUrlToImageCacheConverter.cs [hunk 2]
```diff
@@ -117,8 +119,9 @@ public object Convert(object[] values, Type targetType, object parameter, Cultur
                             imageResult = defaultPackageIcon;
                         }
                     }
-                    catch (Exception)
+                    catch (Exception ex)
                     {
+                        System.Diagnostics.Debug.WriteLine($"msg: {ex.Message} stack: {ex.StackTrace}");
```
Comment (nkolev92) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/Converters/IconUrlToImageCacheConverter.cs]: Remove this.

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Converters/IconUrlToImageCacheConverter.cs (hunk 1 above)
Comment (dominoFire) [reply src/NuGet.Clients/NuGet.PackageManagement.UI/Converters/IconUrlToImageCacheConverter.cs]: Yes. I will remove it.

Comment (dominoFire) [reply src/NuGet.Clients/NuGet.PackageManagement.UI/Converters/IconUrlToImageCacheConverter.cs]: Only in the line below. I left the variable to increase readability and minimize changeset, since extracting the icon entry involves multiple steps/operation calls.

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Converters/IconUrlToImageCacheConverter.cs (hunk 2 above)
Comment (dominoFire) [reply src/NuGet.Clients/NuGet.PackageManagement.UI/Converters/IconUrlToImageCacheConverter.cs]: I will remove it.

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Converters/IconUrlToImageCacheConverter.cs (hunk 1 above)
Comment (dominoFire) [reply src/NuGet.Clients/NuGet.PackageManagement.UI/Converters/IconUrlToImageCacheConverter.cs]: Analyzing the MSIL code, it seems to have less operations if the intermediate variable is removed. So, I will remove `ìconEntry`

Short version:

```asm
// string packageFile = PathUtility.StripLeadingDirectorySeparators(Uri.UnescapeDataString(iconUrl.Fragment).Substring(1));
IL_0130: ldfld class [System]System.Uri NuGet.PackageManagement.UI.IconUrlToImageCacheConverter/'<>c__DisplayClass5_0'::iconUrl
IL_0135: callvirt instance string [System]System.Uri::get_Fragment()
IL_013a: call string [System]System.Uri::UnescapeDataString(string)
IL_013f: ldc.i4.1
IL_0140: callvirt instance string [mscorlib]System.String::Substring(int32)
IL_0145: call string [NuGet.Common]NuGet.Common.PathUtility::StripLeadingDirectorySeparators(string)
IL_014a: stloc.s 15
```

Readable version:

```asm
// string filename = Uri.UnescapeDataString(iconUrl.Fragment).Substring(1);
IL_0130: ldfld class [System]System.Uri NuGet.PackageManagement.UI.IconUrlToImageCacheConverter/'<>c__DisplayClass5_0'::iconUrl
IL_0135: callvirt instance string [System]System.Uri::get_Fragment()
IL_013a: call string [System]System.Uri::UnescapeDataString(string)
IL_013f: ldc.i4.1
IL_0140: callvirt instance string [mscorlib]System.String::Substring(int32)
IL_0145: stloc.s 15
// string packageFile = PathUtility.StripLeadingDirectorySeparators(filename);
IL_0147: ldloc.s 15
IL_0149: call string [NuGet.Common]NuGet.Common.PathUtility::StripLeadingDirectorySeparators(string)
IL_014e: stloc.s 16
```

**Review by nkolev92 — APPROVED** [review]
